File: src/main/java/io/vproxy/vfx/ui/wrapper/ThemeLabel.java
package io.vproxy.vfx.ui.wrapper;

import io.vproxy.vfx.manager.font.FontManager;
import io.vproxy.vfx.theme.Theme;
import javafx.scene.control.Label;

public class ThemeLabel extends Label {
    public ThemeLabel() {
        setTextFill(Theme.current().normalTextColor());
    }

    public ThemeLabel(String text) {
        super(text);
        setTextFill(Theme.current().normalTextColor());
        FontManager.get().setFont(this);
    }
}

File: src-ts/src/ui/wrapper/ThemeLabel.ts
import { Label } from '../../javafx/Label.js';
import { Theme } from '../../theme/Theme.js';
import { FontManager } from '../../manager/font/FontManager.js';

const _NO_ARG = Symbol('ThemeLabel.NO_ARG');

export class ThemeLabel extends Label {
  constructor();
  constructor(text: string);
  constructor(text: string | typeof _NO_ARG = _NO_ARG) {
    super(text === _NO_ARG ? '' : text);
    this.setTextFill(Theme.current().normalTextColor());
    if (text !== _NO_ARG) {
      FontManager.get().setFont(this);
    }
  }
}
